keras_contrib/layers/capsule.py

```diff
@@ -0,0 +1,177 @@
+            self.activation = squash
+        else:
+            self.activation = activations.get(activation)
+
+    def build(self, input_shape):
+        super(Capsule, self).build(input_shape)
+        input_dim_capsule = input_shape[-1]
+        if self.share_weights:
+            self.W = self.add_weight(name='capsule_kernel',
+                                     shape=(1,
+                                            input_dim_capsule,
+                                            self.num_capsule *
+                                            self.dim_capsule),
+                                     initializer='glorot_uniform',
+                                     trainable=True)
+        else:
+            input_num_capsule = input_shape[-2]
+            self.W = self.add_weight(name='capsule_kernel',
+                                     shape=(input_num_capsule,
+                                            input_dim_capsule,
+                                            self.num_capsule *
+                                            self.dim_capsule),
+                                     initializer='glorot_uniform',
+                                     trainable=True)
+
+    def call(self, u_vecs):
+        if self.share_weights:
+            u_hat_vecs = K.conv1d(u_vecs, self.W)
+        else:
+            u_hat_vecs = K.local_conv1d(u_vecs, self.W, [1], [1])
+
+        batch_size = K.shape(u_vecs)[0]
+        input_num_capsule = K.shape(u_vecs)[1]
+        u_hat_vecs = K.reshape(u_hat_vecs, (batch_size,
+                                            input_num_capsule,
+                                            self.num_capsule,
+                                            self.dim_capsule))
+
+        u_hat_vecs = K.permute_dimensions(u_hat_vecs, (0, 2, 1, 3))
+        b = K.zeros_like(u_hat_vecs[:, :, :, 0])
+
+        for i in range(self.routings):
+            c = K.softmax(b, 1)
+            o = K.batch_dot(c, u_hat_vecs, [2, 2])
+            if K.backend() == 'theano' or len(K.int_shape(o)) == 4:
+                o = K.sum(o, axis=1)
+            if i < self.routings - 1:
+                o = K.l2_normalize(o, -1)
+                b = K.batch_dot(o, u_hat_vecs, [2, 3])
+                if K.backend() == 'theano' or len(K.int_shape(b)) == 4:
+                    b = K.sum(b, axis=1)
+
+        return self.activation(o)
+
+    def compute_output_shape(self, input_shape):
+        return (None, self.num_capsule, self.dim_capsule)
+
+    def get_config(self):
+        config = {'num_capsule': self.num_capsule,
+                  'dim_capsule': self.dim_capsule,
+                  'routings': self.routings,
+                  'share_weights': self.share_weights,
+                  'activation': activations.serialize(self.activation)}
+
+        base_config = super(Capsule, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+get_custom_objects().update({'Capsule': Capsule})
```

Comment:
I know all other classes do this, but let's not add it to the custom objects. Users will do it manually. Otherwise we'll end up with conflicts of other python packages do the same.

Reply:
Removed that line.